.github/workflows/build.yml

```diff
@@ -0,0 +1,93 @@
+name: Continuous Integration Checks
+
+on:
+  push:
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        toolchain: [ stable,
+                     beta,
+                     # 1.22.0 is MSRV for rust-lightning in general:
+                     1.22.0,
+                     # 1.34.2 is Debian stable
+                     1.34.2,
+                     # 1.39.0 is MSRV for lightning-net-tokio and generates coverage
+                     1.39.0]
+        include:
+          - toolchain: stable
+            build-net-tokio: true
+          - toolchain: beta
+            build-net-tokio-tokio: true
+          - toolchain: 1.39.0
+            build-net-tokio: true
+            coverage: true
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout source code
+        uses: actions/checkout@v2
+      - name: Install Rust ${{ matrix.toolchain }} toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: ${{ matrix.toolchain }}
+          override: true
+          profile: minimal
+      - name: Build on Rust ${{ matrix.toolchain }} with net-tokio
+        if: ${{ matrix.build-net-tokio }}
+        run: RUSTFLAGS="-C link-dead-code" cargo build --verbose
+      - name: Build on Rust ${{ matrix.toolchain }}
+        if: ${{ matrix.build-net-tokio }} != true
+        run: RUSTFLAGS="-C link-dead-code" cargo build --verbose -p lightning
```

Comment:
I could be mistaken, but it looks like this step is still running on rust-stable, etc where it shouldn't be? And the `Test on Rust` steps also seem to be running redundantly.
https://i.imgur.com/C0HYKL8.png (from your link to the runs in the PR description)

Reply:
Also seems to be the case for 1.39.0

Reply:
Yep, I totally misunderstood the `if:` syntax. Should look good now. Thanks.